server/migrations/2023-08-25-211716_init/up.sql

```diff
@@ -0,0 +1,29 @@
+CREATE TABLE users (
+    id INT PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    name VARCHAR(255) NOT NULL,
+    email VARCHAR(255) NOT NULL,
+    provider VARCHAR(32) NOT NULL,
+    provider_id VARCHAR(255) NOT NULL,
+    created_at DATETIME NOT NULL
+);
+
+CREATE TABLE groups (
+    id INT PRIMARY KEY NOT NULL AUTO_INCREMENT,
+    owner_id INT NOT NULL,
+    created_at DATETIME NOT NULL
+);
+
+CREATE TABLE user_groups (
+   id INT PRIMARY KEY NOT NULL AUTO_INCREMENT,
+   user_id INT NOT NULL,
+   group_id INT NOT NULL,
+   created_at DATETIME NOT NULL
+);
+
+CREATE INDEX users_email_idx ON users (email);
+CREATE INDEX users_provider_id_idx ON users (provider_id);
+
+CREATE INDEX groups_owner_id_idx ON groups (owner_id);
```

Comment:
In the future we might want to have multiple groups - example for work and for family.